.github/workflows/ci.yaml

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This workflow will build a Java project with Maven
+# See also:
+#   https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: CI
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  mvn:
+    strategy:
+      matrix:
+        profile:
+          - name: 'full-build-jdk8'
+            jdk: 8
+            args: '-Pfull-build apache-rat:check verify -DskipTests spotbugs:check checkstyle:check'
+          - name: 'full-build-jdk11'
+            jdk: 11
+            args: '-Pfull-build apache-rat:check verify -DskipTests spotbugs:check checkstyle:check'
+          - name: 'selected-tests'
+            jdk: 11
+            # these tests were somewhat arbitrarily selected to test basic
+            # functionality more tests should run, and possibly in parallel, as
+            # the testing situation improves in ZooKeeper
+            args: '-Pfull-build verify -Dsurefire-forkcount=1 -DskipCppUnit -DfailIfNoTests=false -Dtest=ClientCnxnSocketTest,ClientSSLTest,ClientTest,QuorumDigestAuthTest,QuorumKerberosAuthTest,QuorumPeerMainTest,QuorumPeerTest,QuorumServerTest,QuorumTest,QuorumZxidSyncTest,SaslAuthTest,SaslClientTest,StatCommandTest,StatTest,WatcherFuncTest,WatcherTest,ZKClientConfigTest,ZooKeeperServerMainTest,ZooKeeperServerTest,ZooKeeperTest'
```

Comment:
can we run the whole suite ?
We must rely on PR validation in order to see that there is no regression, there is no "minimum acceptance test"

Reply:
I was unable to get the full test suite to pass as it is today. This PR would lay the groundwork for getting all the tests to pass in GitHub Actions and eventually replace the Jenkins precommit job.

However, today, it only replaces the `x86_64` architecture Linux builds in Travis CI (which didn't execute any tests).

Getting all the tests to pass is too much for one PR, and will need to happen incrementally once this is in place and contributors can use it to watch for flaky or problematic tests.

Reply:
Okay, it looks like we're closer to getting *all* tests to pass in GH Actions than I previously thought... I just tried to build all of them in a separate branch, and only had two test failures, and the entire build completed in 70 minutes (using `surefire-forkcount=1`), which is comparable to what I got on my laptop, where I got all tests to pass after 2 tries.

* surefire Java test: `RequestThrottlerTest.testGlobalOutstandingRequestThrottlingWithRequestThrottlerDisabled:340 expected: <3> but was: <6>`
* zktest-mt (C client): `TestReconfigServer::testNonIncrementalStarting zookeeper ... FAILED TO START`  
  previous line was: ``zktest-mt: /home/runner/work/zookeeper/zookeeper/zookeeper-client/zookeeper-client-c/tests/ZooKeeperQuorumServer.cc:67: void ZooKeeperQuorumServer::start(): Assertion `system(command.c_str()) == 0' failed.``

Reply:
I will update this PR to run all the tests.